crates/dht/src/dht/tests/harness.rs

```diff
@@ -0,0 +1,429 @@
+use crate::arc_set::ArcSet;
+use crate::dht::snapshot::DhtSnapshot;
+use crate::dht::tests::SECTOR_SIZE;
+use crate::{Dht, DhtSnapshotNextAction};
+use kitsune2_api::{
+    AgentId, DhtArc, DynOpStore, K2Result, OpId, OpStore, StoredOp, Timestamp,
+};
+use kitsune2_memory::{Kitsune2MemoryOp, Kitsune2MemoryOpStore};
+use rand::RngCore;
+use std::collections::HashMap;
+use std::sync::Arc;
+
+/// Intended to represent a single agent in a network, which knows how to sync with
+/// some other agent.
+pub(crate) struct DhtSyncHarness {
+    pub(crate) store: Arc<Kitsune2MemoryOpStore>,
+    pub(crate) dht: Dht,
+    pub(crate) arc: DhtArc,
+    pub(crate) agent_id: AgentId,
+    pub(crate) discovered_ops: HashMap<AgentId, Vec<OpId>>,
+}
+
+pub(crate) enum SyncWithOutcome {
+    InSync,
+    CannotCompare,
+    SyncedDisc,
+    SyncedRings,
+    DiscoveredInSync,
+}
+
+impl DhtSyncHarness {
+    pub(crate) async fn new(current_time: Timestamp, arc: DhtArc) -> Self {
+        let store = Arc::new(Kitsune2MemoryOpStore::default());
+        let dht = Dht::try_from_store(current_time, store.clone())
+            .await
+            .unwrap();
+
+        let mut bytes = [0; 32];
+        rand::thread_rng().fill_bytes(&mut bytes);
+        let agent_id = AgentId::from(bytes::Bytes::copy_from_slice(&bytes));
```

Comment:
There's a `random_agent_id` fn in the `core_fetch` tests which I invite you to avail of here.